functorch/_src/eager_transforms.py

```diff
@@ -324,14 +322,89 @@ def wrapper(*args, **kwargs):
         return grad_input, output
     return wrapper
 
-def grad(f, argnums=0, has_aux=False):
-    @wraps(f)
+def grad(func: Callable, argnums: argnums_t = 0, has_aux: bool = False) -> Callable:
+    """``grad`` operator helps computing gradients of the output of a function.
+    ``grad(func)`` returns a new function that computes the gradients of ``func``
+    with respect to the input. This operator can be used multiple times to compute
+    higher-order gradients.
+
+    Args:
+        func (Callable): A Python function that takes one or more arguments.
+            Must return a single-element Tensor. If specified ``has_aux=True``, function can
+            return a tuple of single-element Tensor and other auxiliairy objects:
+            ``(output, aux)``.
+        argnums (int or Tuple[int]): Specifies arguments to compute gradients with respect to.
+            ``argnums`` can be single integer or tuple of integers. Default: 0.
+        has_aux (bool): Flag indicating that ``func`` returns a tensor and other
+            auxiliairy objects: ``(output, aux)``. Default: False.
+
+    Returns:
+        Function to compute gradients on its inputs. By default, the output of
+        the function is gradients tensor. If specified ``has_aux=True``, tuple of gradients
+        and output auxiliairy objects is returned.
+        If ``argnums`` is a tuple of integers, a tuple of output gradients with respect to
+        each ``argnums`` value is returned
+
+    Example of using ``grad``:
+
+        >>> from functorch import grad
+        >>> x = torch.randn([])
+        >>> cos_x = grad(lambda x: torch.sin(x))(x)
+        >>> assert torch.allclose(cos_x, x.cos())
+        >>>
+        >>> # Second-order gradients
+        >>> neg_sin_x = grad(grad(lambda x: torch.sin(x)))(x)
+        >>> assert torch.allclose(neg_sin_x, -x.sin())
+
+    When composed with ``vmap``, ``grad`` can be used to compute per-sample-gradients:
+
+        >>> from functorch import vmap
+        >>> batch_size, feature_size = 3, 5
+        >>>
+        >>> def model(weights,feature_vec):
```

Comment:
nit: space in between parameters
```suggestion
        >>> def model(weights, feature_vec):
```